### ISSUE TITLE
Cannot open game after export, if contains other languages characters

### ISSUE THREAD

**Issue by XHXIAIEIN** [Scirra/Construct-bugs#7787]
## Problem description

Added:
This issue may be caused by not fil in the `ID` in the property field.

![image](https://github.com/Scirra/Construct-bugs/assets/45864744/270bb36a-020c-4486-b15a-d6ea3b6924bb)

So, the issue report will become to:
When exporting, the dialog for verifying whether the ID is filled in is missing.

`package.json` : the'name' is empty so it can not open game.

![acf1665f90699c7e580c2c1bfc26cf34](https://github.com/Scirra/Construct-bugs/assets/45864744/0b99c6f0-304b-41dc-9aa4-624247e4bb30)

----

In C3 editor Property, if contains other languages characters, it cannot open game after export.
For example, fill in a chinese character as Project Name

```
新项目
```

![8bc3154d231ca54c159ff619c9a35d3c](https://github.com/Scirra/Construct-bugs/assets/45864744/2a7dc1d8-fc90-44d8-aa7c-b014f58b2db1)

This is also replicated in English on the editor language
![Snipaste_2024-01-29_03-51-50](https://github.com/Scirra/Construct-bugs/assets/45864744/b809ff78-14d2-4e83-a978-1512b8c413c9)

This can be replicated in both NW.js and Webview2

![image](https://github.com/Scirra/Construct-bugs/assets/45864744/29f4e1d9-c1d7-40fc-b406-344edf207fea)

![image](https://github.com/Scirra/Construct-bugs/assets/45864744/19cd627c-3728-4202-a357-a08950aa8613)

## Attach a .c3p

Project file.
[issue-webview2-language-support.c3p.zip](https://github.com/Scirra/Construct-bugs/files/14077492/issue-language-support.c3p.zip)

## Steps to reproduce

1.
2.
3.

## More details

**Affected browsers/platforms:** 

**First affected release:** 

## System details

<details><summary>View details</summary>

Platform information
Product: Construct 3 r376 (beta)
Browser: Chrome 121.0.6167.85
Browser engine: Chromium
Context: browser
Operating system: Windows 11
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 16
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
Language setting: en-US

Local storage
Storage quota (approx): 477 gb
Storage usage (approx): 758 mb (0.2%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: Google Inc. (AMD)
Renderer: ANGLE (AMD, AMD Radeon(TM) Graphics (0x00001681) Direct3D11 vs_5_0 ps_5_0, D3D11)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_depth_clamp
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_draw_buffers_indexed
OES_texture_float_linear
OVR_multiview2
WEBGL_clip_cull_distance
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
WEBGL_polygon_mode
WEBGL_provoking_vertex
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm;codecs=opus)
WebM Vorbis (audio/webm;codecs=vorbis)
MPEG-4 Opus (audio/mp4;codecs=opus)
MPEG-4 AAC (audio/mp4;codecs=mp4a.40.2)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav;codecs=1)
Supported encode formats:

WebM Opus (audio/webm;codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm;codecs=av01.0.00M.08)
WebM VP9 (video/webm;codecs=vp9)
WebM VP8 (video/webm;codecs=vp8)
MPEG-4 AV1 (video/mp4;codecs=av01.0.00M.08)
MPEG-4 H.265 (video/mp4;codecs=hev1.1.2.L93.B0)
MPEG-4 H.264 (video/mp4;codecs=avc1.420034)
Supported encode formats:

WebM AV1 (video/webm;codecs=av01.0.00M.08)
WebM VP9 (video/webm;codecs=vp9)
WebM VP8 (video/webm;codecs=vp8)
WebM H.264 (video/webm;codecs=avc1.420034)

</details>

**Comment by XHXIAIEIN** [comment]
## Example 2

If Sprite Object contains other languages characters, export the game, after play the loader page, it will enter a black screen.

![Snipaste_2024-01-29_23-42-49](https://github.com/Scirra/Construct-bugs/assets/45864744/e2bdb8dc-cacb-482e-8650-bdcb2763bbc6)

[issue-chinese-name.c3p.zip](https://github.com/Scirra/Construct-bugs/files/14086432/issue-chinese-name.c3p.zip)


![Snipaste_2024-01-29_23-46-31](https://github.com/Scirra/Construct-bugs/assets/45864744/cf9d43fe-25a2-4eae-a14b-0899956b1f9d)

**Comment by AshleyScirra** [comment]
The project link in the original issue doesn't work for me. I was able to reproduce it though: [issue7787.zip](https://github.com/Scirra/Construct-bugs/files/14087933/issue7787.zip)

With NW.js, the documentation seems to have strict requirements on the package name. Construct uses the project name if possible, else the project ID, but if neither are valid then it's empty and fails to export. The NW.js documentation says the package name must be "alphanumeric" which usually means only A-Z and 0-9 (i.e. English characters only) and it does indeed seem to have unicode encoding bugs if given a Chinese package name, so that seems best avoided. Instead I've changed the next beta to show a warning and require a valid project ID to be set if the project name is not valid for NW.js.

For Windows WebView2, it handles unicode names just fine actually, but imposed the same restriction as NW.js unnecessarily. I've removed the restriction for the next beta.

The exported project works fine for me on my local web server. It could be your server is not handling unicode properly. It's also unrelated to the original issue so would need to be followed up separately anyway.
